cmd/clusterctl/client/config/imagemeta_client_test.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"testing"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/test"
+)
+
+func Test_imageMetaClient_AlterImage(t *testing.T) {
+	type fields struct {
+		reader Reader
+	}
+	type args struct {
+		component string
+		image     string
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    string
+		wantErr bool
+	}{
+		{
+			name: "no image config, image should not be changes",
+			fields: fields{
+				reader: test.NewFakeReader(),
+			},
+			args: args{
+				component: "any",
+				image:     "us.gcr.io/k8s-artifacts-prod/cluster-api/cluster-api-controller:v0.2.10",
+			},
+			want:    "us.gcr.io/k8s-artifacts-prod/cluster-api/cluster-api-controller:v0.2.10",
+			wantErr: false,
+		},
+		{
+			name: "image config for cert-manager, image for the cert-manager should be changed",
+			fields: fields{
+				reader: test.NewFakeReader().WithImageMeta("cert-manager", "foo-repository.io", "foo-tag"),
+			},
+			args: args{
+				component: "cert-manager",
+				image:     "us.gcr.io/k8s-artifacts-prod/cluster-api/cluster-api-controller:v0.2.10",
```

Comment:
To be more realistic, change the image to be `cert-manager`?

Reply:
done